### ISSUE TITLE
Seem not work with styled-components

### ISSUE THREAD

**Issue by linonetwo** [citycide/param.macro#12]
This works
```js
const getColor = it.color
const ActionName = styled.div`
  text-align: center;
  font-weight: bold;
  font-size: 16px;
  color: ${getColor};
  border: 1px solid ${getColor};
  padding: 5px;
`;
// same as
const ActionName = styled.div`
  text-align: center;
  font-weight: bold;
  font-size: 16px;
  color: ${({ color }) => color};
  border: 1px solid ${({ color }) => color};
  padding: 5px;
`;
```
And this don't:
```js
const ActionName = styled.div`
  text-align: center;
  font-weight: bold;
  font-size: 16px;
  color: ${it.color};
  border: 1px solid ${it.color};
  padding: 5px;
`;
```
Any possible reason?

**Comment by citycide** [comment]
@linonetwo which one are you importing first and have you tried to switch them? Remember that the order of imports matters for macros.

**Comment by linonetwo** [comment]
Do you think expressions in TemplateLiteral can be regarded as a kind of assignment?

Since if a function being passed to a TemplateLiteral, the only reason is it's inside a TaggedTemplateExpression. So expressions in TemplateLiteral will be assign to tag function's arguments.

**Comment by citycide** [comment]
@linonetwo potentially, and that would be the easy fix in this case. But I'd like to try and avoid coming across more and more scenarios that need to be special cased if we'd instead be able to have a consistent strategy. Making `it` be even more "in place" than it is now might be in order.

**Comment by lorefnon** [comment]
It might be useful to expose a pseudo function that defines the function boundary. 

```
import {$, it} from 'param.macro';

const fn = fn2($(it.prop1), $(it.prop2))
```

Compiles to: 

```
const fn = fn2(
    (it) => it.prop1,
    (it) => it.prop2
)
```

**Comment by citycide** [comment]
@linonetwo I've thought about this for a while and think you're spot on &mdash; ``` fn`...` ``` is just a function call and is the only alternative function call syntax there really is, so rather than over-engineer this we should probably just stop traversal within this template expression (called a "quasi" by the spec I think).

**Comment by citycide** [comment]
Anyone have thoughts on whether this should be considered a breaking change and released in `3.0.0` or is ok to release as a `2.x` feature release?

**Comment by citycide** [comment]
[v3.0.0](https://github.com/citycide/param.macro/releases/tag/v3.0.0) should handle this as you expect!
